stripped/colabs/Regresja_logistyczna_i_SVM_stripped.py

```diff
@@ -0,0 +1,1180 @@
+full_winter_data = full_winter_data.drop(columns=['Month', 'Year', 'lon', 'lat'])
+
+corr = full_winter_data.corr(method = "pearson")
+cmap = sns.diverging_palette(250, 323, 80, 60, center='dark', as_cmap=True)
+sns.heatmap(corr, vmax =1, vmin=-0.3, cmap = cmap, square = True, linewidths = 0.2, annot=True)
+
+sns.pairplot(data=full_winter_data, diag_kind='kde')
+
+def Kfolds_crossvalidation( model: BaseEstimator, folds: np.ndarray, folds_labels: List[List[int]]) -> dict:
+
+ validation_accuracy = []
+ validation_recall = []
+ validation_precision = []
+ validation_ROCAUC = []
+ validation_jaccard = []
+
+ k=len(folds)
+ for i in range(k):
+       X_test = folds[i]
+       y_test = folds_labels[i]
+       train_folds = [fold for fold in folds if fold.tolist() != folds[i].tolist()]
+       train_labels = [labels for labels in folds_labels if labels != folds_labels[i]]
+       X_train = np.concatenate(train_folds)
+       y_train = np.concatenate(train_labels)
+       trained_model = model.fit(X_train, y_train)
+       y_pred = trained_model.predict(X_test)
+       validation_accuracy.append(accuracy_score(y_test, y_pred))
+       validation_recall.append(recall_score(y_test, y_pred))
+       validation_precision.append(precision_score(y_test, y_pred, zero_division=0))
+       validation_ROCAUC.append(roc_auc_score(y_test, y_pred))
+       validation_jaccard.append(jaccard_score(y_test, y_pred))
+
+       results = {'accuracy': mean(validation_accuracy),
+                 'recall': mean(validation_recall),
+                 'precision': mean(validation_precision),
+                  'ROC-AUC': mean(validation_ROCAUC),
+                  'jaccard_score': mean(validation_jaccard)}
+
+ return results
+
+def data_standarization(df: pd.DataFrame) -> np.ndarray:
+    ''' the function performs data standarization
+        on data stored in given pandas DataFrame '''
+
+    scaler = preprocessing.StandardScaler()
+    standarized = scaler.fit_transform(df)
+
+    return standarized
+
+def GridSearchCV_LogisticRegression(folds: np.ndarray, folds_labels: List[List[int]], param_grid: dict) -> pd.DataFrame:
+  all_res = dict({'penalty': [], 'solver': [], 'class_weight': [], 'C': [], 'accuracy': [], 'recall': [],
+                      'precision': [], 'ROC-AUC': [], 'jaccard_score': []})
+
+  for penalty in param_grid['penalty']:
+    for weight in param_grid['class_weight']:
+      for solver in param_grid['solver']:
+         for c in param_grid['C']:
+            model = LogisticRegression(penalty=penalty, class_weight = weight, solver=solver, C=c, max_iter=10000)
+            results = {'penalty': penalty, 'solver': solver, 'class_weight': str(weight), 'C': c}
+            crossval_results = Kfolds_crossvalidation(model, folds, folds_labels)
+            results.update(crossval_results)
+            results_sorted = dict({key : results[key] for key in all_res.keys()})
+            all_res = {key: all_res[key] + [results_sorted[key]] for key in all_res}
```

Comment:
Tu nie bardzo wiem o co chodzi. Wyjaśnisz. Bo np. 
```
dict({key : results[key] for key in all_res.keys()})
```
to to samo co
```
{key : results[key] for key in all_res.keys()}
```

Reply:
Pomyłka. Poprawione.